code/modules/mob/living/carbon/superior_animal/robots/subtype/mining/mining_drone.dm

```diff
@@ -0,0 +1,130 @@
+	to_chat(user, SPAN_NOTICE("	Ore Gathering : [pickup ? "Active" : "Not Active"]"))
+	to_chat(user, SPAN_NOTICE("	Mining : [mining ? "Active" : "Not Active"]"))
+
+/mob/living/carbon/superior_animal/robot/mining/Life()
+	..()
+
+	if(!look_around() && wandering) // If we didn't find anything of value and can wandering
+		target = pick(oview(viewRange + 1, src)) // Go somewhere random
+
+	// Are we going to a minable turf despite not being supposed to?
+	if(istype(target, /turf/simulated/mineral && !mining))
+		target = null // reset the target
+
+	// Are we going to ores on the ground despite not being supposed to?
+	if(istype(target, /obj/item/ore && !pickup))
+		target = null // reset the target
+
+	if(target) // Do we have a destination?
+		walk_to(src, target, 1, move_to_delay) // Go there
+	else
+		walk_to(src, 0) // Or else stop
+
+/mob/living/carbon/superior_animal/robot/mining/death()
+	drop_loot()
+	..()
+
+/mob/living/carbon/superior_animal/robot/mining/attack_hand(mob/user as mob)
+	if(user.a_intent == I_HELP) // Are we on help intent?
+		var/possible_choices = list("Wandering", "Ore Gathering", "Ore Mining")
+		var/choice = input(user, "Which parameter do you want to toggle in the mining bot?", "Parameters", null) as null|anything in possible_choices
+		switch(choice)
+			if("Wandering")
+				wandering = !wandering
+				to_chat(user, "[src] will [wandering ? "" : "no longer"] wander.")
+			if("Ore Gathering")
+				pickup = !pickup
+				to_chat(user, "[src] will [pickup ? "" : "no longer"] take ore on the ground.")
+			if("Ore Mining")
+				mining = !mining
+				to_chat(user, "[src] will [mining ? "" : "no longer"] mine ore.")
+	else ..()
+
+/mob/living/carbon/superior_animal/robot/mining/attackby(obj/item/W as obj, mob/user as mob)
+	var/obj/item/T // Define the tool variable early on to avoid compilation problem and to allow us to use tool-unique variables
+	if(user.a_intent == I_HELP) // Are we helping ?
+		// If it is a tool, assign it to the tool variable defined earlier.
+		if(istype(W, /obj/item/tool))
+			T = W
+
+		if(QUALITY_PULSING in T.tool_qualities) // We pulse the drone to get the loot
+			drop_loot() // Drop the loot
+			return
+
+	// If nothing was ever triggered, continue as normal
+	..()
+
+/mob/living/carbon/superior_animal/robot/mining/proc/look_around()
+	for(var/O in oview(1, src)) // Check our surroundings.
+		if(mining && istype(O, /turf/simulated/mineral)) // Is it a turf?
+			var/turf/simulated/mineral/M = O
+			if(M.mineral) // Does it have stuff to mine?
+				mine(M) // Mine the turf
+				target = null
+				. = TRUE // We'll return later, keep looping.
+				continue
+
+		if(pickup && istype(O, /obj/item/ore)) // Is it ore on the ground?
+			var/obj/item/ore/Ore = O
+			pick_ore(Ore) // Pick it up
+			target = null
+			. = TRUE // We'll return later, keep looping.
+			continue
+
+	if(.) // Did we mine or pick up stuff earlier on?
```

Comment:
Bit of a weird usage for the '.', but okay- might've been cleaner to set up a boolean varable, but I'm also the sort to give literally every variable a long and descriptive name.

Reply:
I mean `.` is the default return variable. If I just do `return`, then what actually happen is `return .`, and `.`'s value by default is null. I'm using it to prevent returning early so that the bot can mine *every* turf and pick up *every* item that it can in one swoop instead of picking up 1 item every 5 seconds.